api/pkg/apis/v1alpha1/providers/states/k8s/k8s_test.go

```diff
@@ -77,112 +80,129 @@ func TestUpsert(t *testing.T) {
 	if testK8s == "" {
 		t.Skip("Skipping because TEST_K8S_STATE enviornment variable is not set")
 	}
+	err := checkActivationCRDApplied()
+	assert.Nil(t, err)
 	provider := K8sStateProvider{}
-	err := provider.Init(K8sStateProviderConfig{
+	err = provider.Init(K8sStateProviderConfig{
 		InCluster:  false,
 		ConfigType: "path",
 	})
 	assert.Nil(t, err)
 	id, err := provider.Upsert(context.Background(), states.UpsertRequest{
 		Value: states.StateEntry{
-			ID: "s123",
```

Comment:
We should keep Target state operation tests as well.

Reply:
Sounds good. I have put them back.